source/docs/hardware/hardware-safety/io_safety.rst

```diff
@@ -0,0 +1,27 @@
+Input/Output Safety Mechanisms Built into the `FRC` Control System
+============================================================================
+
+roboRIO Control System
+^^^^^^^^^^^^^^^^^^^^^^
+
+There are multiple safety mechanisms on the robot that handle input / output operations while it is powered on.
+
+Robot side: there are multiple hardware and software components involved. Outputs of the RoboRIO (e.g. :term:`PWM` s) are controlled by the :term:`FPGA` hardware. :term:`NetComm` is a software daemon that talks to the DS, the FPGA, and the user program. Inside of the user process \(the team\’s robot program\), there\’s a NetComm :term:`DLL` component that talks to the FPGA, CAN, and the NetComm daemon. And of course there are CAN motor controllers on the CAN bus.
+
+- The FPGA has a system :term:`watchdog`. This watchdog will time out and force a “disable” of PWM motor outputs if NetComm hasn\’t told it it\’s received an enable packet in the last 125 `ms`.
```

Comment:
```suggestion
- The FPGA has a system :term:`watchdog`. This watchdog will time out and force a “disable” of PWM motor outputs if NetComm hasn\’t told it it\’s received an enable packet in the last 125 `ms`.
```

Don't use non-UTF8 quotations. Replace “ with ". This helps our translators. Additionally, replace the non-UTF8 apostrophe with '